addon/-private/class/modifier-manager.ts

```diff
@@ -0,0 +1,70 @@
+import Ember from "ember";
+import { capabilities } from "@ember/modifier";
+import { set } from "@ember/object";
+import { schedule } from "@ember/runloop";
+
+import ClassBasedModifier, {
+  DESTROYING,
+  DESTROYED,
+  InTeardown,
+} from "./modifier";
+import { ModifierArgs } from "ember-modifier/-private/interfaces";
+
+class ClassBasedModifierManager {
+  capabilities = capabilities("3.13");
+
+  createModifier(
+    factory: { owner: unknown; class: typeof ClassBasedModifier },
+    args: ModifierArgs
+  ) {
+    let { owner, class: modifier } = factory;
```

Comment:
This is kinda wild, we should not be getting the `owner` from this argument 🤔 

See the modifier manager API RFC https://github.com/emberjs/rfcs/blob/master/text/0373-Element-Modifier-Managers.md#createmodifier, which says:

> The first argument passed to createModifier is the result returned from the factoryFor API. It contains a class property, which gives you the the raw class (the default export from app/modifiers/foo.js) and a create function that can be used to instantiate the class with any registered injections, merging them with any additional properties that are passed.

---

Since this is a carry over from existing implementation I think an inline comment with a link to an issue is fine for now. Ember should also be updated to _not_ pass the `owner` like this (though that will have to go through a deprecation cycle).

cc @pzuraq

Reply:
Tracking at #25